source/MaterialXTest/MaterialXGenGlsl/GenGlsl.cpp

```diff
@@ -18,6 +18,12 @@ namespace mx = MaterialX;
 
 TEST_CASE("GenShader: GLSL Syntax Check", "[genglsl]")
 {
+    mx::GenContext context(mx::GlslShaderGenerator::create());
+    mx::FileSearchPath searchPath = mx::getDefaultDataSearchPath();
+    mx::DocumentPtr libraries = mx::createDocument();
+    mx::loadLibraries({ "libraries" }, searchPath, libraries);
+    context.getShaderGenerator().registerTypeDefs(libraries, context);
```

Comment:
Nitpick: Perhaps you could add a wrapper method, `GenContext::registerTypeDefs( DocumentPtr )`, that internally call its shader generator. Just to avoid the "recursive" passing of the context here that looks a bit strange.

So the call would instead be: `context.registerTypeDefs(libraries);`